examples/example-oauth/src/main/java/io/grpc/examples/oauth/ExampleOAuth2Credentials.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.examples.oauth;
+
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.OAuth2Credentials;
```

Comment:
We are relying on Google client APIs here. As a user this would make me think that either:

- gRPC only supports oauth if I couple my app with Google libraries
- This example is specific to Google libraries and I don't have a good example of what to do if I don't use Google libraries

Explain somewhere (maybe in README.md) what the deal is?

Reply:
Good point. @ejona86 and I talked about it. I'll add some explanation in the README

Reply:
done